enhancements/network-admission/allow-external-ip-overrides.md

```diff
@@ -0,0 +1,161 @@
+However, users who would upgrade from 4.1 z-stream where admin over-ride of external IPs is not just supported but needed 
+to 4.2, would expect their service configurations to work as is. This proposal therefore is to allow the admin over-ride 
+in 4.x as a whole.
+
+### Goals
+
+- Provide a mechanism for admin user(s) to over-ride cluster configured values of ranges of external IPs for a service. 
+<a name="admin-user">An admin user here is defined as someone who is allowed to create a subresource under a service in a given namespace </a>.
+
+- As a corollary, non-admin user should not be able to patch/update the service spec with external IPs that fall out of the 
+range specified at the cluster level.
+
+- Both admin and non-admin users should still be able to specify external IPs for a service that fall within the range 
+specified at the cluster level.
+
+### Non-Goals
+
+- Escalation of privileges by causing service traffic to be routed to a malicious external IP address.
+
+## Proposal
+
+- As an administrator with [sufficient privilege](#admin-user) on a 4.x Openshift cluster, I want the ability to 
+specify external IPs which may fall out of the range specified by cluster administrators for services belonging to my app.
+
+This proposal is to add an RBAC check in case the external IPs specified for the service don't fall in the range of the 
+cluster specified **externalIpNetworkCIDR**. This is needed particularly in 4.1, where the external IPs will be blocked 
+completely. In case, there are validation errors as a result of the specified external IP(s) not being in the valid range, 
+the RBAC checker will validate that the user has permissions to create a dummy subresource for the service in question. 
+If the result of that check is an Allow, the user is allowed to set the external IP(s) on the service.
+
+Code snippet that will perform a RBAC check if the IP falls outside the allow range or is in the reject range:
+
+`errs` below is a list of errors obtained after doing a range check for the specified external IP.
+
+	``` go
+    if len(errs) > 0 {
+            		//if there are errors reported, resort to RBAC check to see
+            		//if this is an admin user who can over-ride the check
+            		allow, err := r.checkAccess(a)
+            		if err != nil {
+            			return err
+            		}
+            		if !allow {
+            			return admission.NewForbidden(a, errs.ToAggregate())
+            		}
+    }
+	```
+            
+Code that performs the actual RBAC check:
+
+	``` go
+    func (r *externalIPRanger) checkAccess(attr admission.Attributes) (bool, error) {
+            	authzAttr := authorizer.AttributesRecord{
+            		User:            attr.GetUserInfo(),
+            		Verb:            "create",
+            		Namespace:       attr.GetNamespace(),
+            		Resource:        "service",
+            		Subresource:     "externalips",
+            		APIGroup:        kapi.GroupName,
```

Comment:
This check is specific to openshift, I recommend using an openshift group.  `network.openshift.io`